scripts/compose-init.sh

```diff
@@ -1,37 +1,99 @@
 #!/bin/bash
+#
+# Check the output from this script on the EC2 VM with:
+#
+#   journalctl -u cloud-final
+#
+
+declare -r SOLR_DATA_DEVICE=${solr_data_device_name}
+
+#
+# Add and configure users.
+#
 
 # Add SSH public key if var was set
 if [[ -n "${ec2_public_key}" ]]; then
-    # But first ensure existance and correct permissions
-    sudo -Hu ec2-user bash <<- EOF
-	umask 0077
-	mkdir -p /home/ec2-user/.ssh
-	touch /home/ec2-user/.ssh/authorized_keys
-	EOF
-    echo "${ec2_public_key}" >> /home/ec2-user/.ssh/authorized_keys
+    install -d -m 0755 -o ec2-user -g ec2-user ~ec2-user/.ssh
+    touch ~ec2-user/.ssh/authorized_keys
+    chown ec2-user: ~ec2-user/.ssh/authorized_keys
+    chmod 0644 ~ec2-user/.ssh/authorized_keys
```

Comment:
Would it make sense to collapse these into a single `install` comand like in the for ec2_users loop?
```suggestion
    install -m 0644 -o ec2-user -g ec2-user /dev/null ~ec2-user/.ssh/authorized_keys
```